lib/librato-sidekiq/middleware.rb

```diff
@@ -62,30 +62,38 @@ def call(worker_instance, msg, queue, redis_pool = nil)
         return result unless enabled
         # puts "#{worker_instance} #{queue}"
 
+        # From the docs (https://github.com/mperham/sidekiq/wiki/Job-Format):
+        # Note that enqueued_at isn't added to the payload for scheduled jobs.
+        # They get the enqueued_at field when they are pushed onto a queue.
+        enqueued_at = msg['enqueued_at'] || start_time
+        latency = (start_time - enqueued_at).to_f
```

Comment:
I'd definitely expect `nil` latency if we don't know it?

Reply:
A scheduled job will have `at` set as the time we wanted it to run.  Perhaps we could use that instead of `start_time` (see: https://github.com/mperham/sidekiq/wiki/Job-Format#scheduled-jobs).

I haven't dug into how the middleware is used though, when is this `call` actually triggered and do we really need to worry about it being called on scheduled jobs that haven't been enqueued yet?

Reply:
Turns out,  don't need to dig too much.  In `self.reconfigure` above we can see we use this middleware as server middleware, and server middleware is wrapped around jobs so by my understanding, all the job payloads that come through here will have been enqueued so I don't think we need to worry about a missing enqueued at.  (See:  https://github.com/mperham/sidekiq/wiki/Middleware#server-middleware)

However, we _might_ want to worry about `at` vs. `enqueued_at` to measure the latency between us asking for a job to be performed and it getting onto a queue (and then we compare that to `start_time` to get the latency between it being on the queue and us actually running it).

Reply:
As @h-lame said, all jobs would be enqueued at this point. Following a `.perform_async` or a `.perform_in(x.seconds)`, the payload is similar and the latency was not affected by the delay in putting the job in the queue.

My main questions are, do you think I should be defensive here?
* If not, I can just assume `msg['enqueued_at']` is always present. 
* If yes, what is the fallback? Do we report latency? Or do we assume it's 0? I got the impression that @AgentAntelope would prefer that we don't report latency at all?

Reply:
Not suggesting we don't report, `nil` implies it doesn't exist or can't be returned, I don't want to accidentally lower averages etc

Reply:
I think we won't need to worry about missing `enqueued_at`, but I also agree with @AgentAntelope, if we couldn't detect `enqueued_at` it would be nice not to report latency because we don't know it, rather than reporting a "guess" using `start_time`.  Perhaps if we don't have `enqueued_at` we should increment a librato counter for that (using the worker class as a source maybe?)

Reply:
I didn't do the counter thing, but if we think we're missing some data we can add it later.
Let me know if you're happy with this in the current state, I have stopped reporting latency if we don't have `enqueued_at`